src/main/java/gov/nasa/jpf/constraints/solvers/nativez3/NativeZ3Solver.java

```diff
@@ -24,112 +27,155 @@
-            Global.setParameter(o.getKey(),o.getValue());
-        }
-        
-        try {
-            this.ctx = new Context(cfg);
-            defaultContext = createContext();
-        } catch (Z3Exception ex) {
-            if (ctx != null) {
-                try {
-                    ctx.dispose();
-                } catch (Throwable t) {
-                }
-            }
-            throw new RuntimeException(ex);
-        }
-    }
-
-    Context getContext() {
-        return ctx;
-    }
-
-    public void dispose() {
-        defaultContext.dispose();
-        defaultContext = null;
-        ctx.dispose();
-        ctx = null;
-    }
-
-    protected void finalize() throws Throwable {
-        super.finalize();
-        if (ctx != null) {
-            dispose();
-        }
-    }
-
-    @Override
-    public Result solve(Expression<Boolean> f, Valuation result) {
-        try {
-            defaultContext.push();
-            defaultContext.add(f);
-            return defaultContext.solve(result);
-        } finally {
-            defaultContext.pop();
-        }
-    }
-
-    @Override
-    public NativeZ3SolverContext createContext() {
-        Solver solver = null;
-
+    BoolExpr z3Expr = rootGenerator.generateAssertion(expr);
+    goal.add(z3Expr);
+    
+    ApplyResult res = tactic.apply(goal);
+    Goal[] subgoals = res.getSubgoals();
+    System.out.println("gov.nasa.jpf.constraints.solvers.nativez3.NativeZ3Solver.eliminateQuantifiers()");
+    Expression result = null;
+    NativeZ3TojConstraintConverter converter = new NativeZ3TojConstraintConverter();
+    for(Goal g: subgoals){
+      BoolExpr[] formulas = g.getFormulas();
+      for(BoolExpr f: formulas){
+        System.out.println(f.toString());
+        System.out.println("gov.nasa.jpf.constraints.solver.ExpressionConversionTest.testExpressionConversion() - asdfasfagdsgasdgas");
+        Expression<Boolean> jConstraintExpr = converter.parse(f);
+        result = result== null? jConstraintExpr :
+                ExpressionUtil.and(result, jConstraintExpr);
         try {
-            solver = ctx.mkSolver();
-            NativeZ3ExpressionGenerator root;
-
-            if (timeout > 0) {
-            Params p = ctx.mkParams();
-                //p.add("timeout",timeout);
-                //p.add(":timeout", timeout); 
-                p.add("timeout", timeout);
-                solver.setParameters(p);     
-            }
-
-            root = new NativeZ3ExpressionGenerator(ctx, solver);
-
-            return new NativeZ3SolverContext(solver, root);
-        } catch (Z3Exception ex) {
-            if (solver != null) {
-                try {
-                    solver.dispose();
-                } catch (Throwable t) {
-                }
-            }
-            throw new RuntimeException(ex);
+          result.print(System.out);
+          System.out.println("parsed printed");
+        } catch (IOException ex) {
```

Comment:
Where does this exception originate? Is it OK to be left un-handled? Can it affect correctness of the returned result?